src/ds++/Assert.hh

```diff
@@ -332,12 +332,11 @@ std::string verbose_error(std::string const &message);
 //------------------------------------------------------------------------------------------------//
 // Always on
 //------------------------------------------------------------------------------------------------//
-#if ( defined __NVCC__  && defined USE_CUDA )
-#define Insist(c, m) if(!(c)) rtt_dsxx::no_exception_insist( #c, m, __FILE__, __LINE__)
-#else
+#if ( defined __CUDA_ARCH__  && defined USE_CUDA )
+#define Insist_device(c, m) if(!(c)) rtt_dsxx::no_exception_insist( #c, m, __FILE__, __LINE__)
+#endif
```

Comment:
What do you think about adding an `else` here that defines `Insist_device` to be the CPU version? That way if I change the shared CPU/GPU code in Jayenne to use `Insist_device` they will still get the correct throwing version in CPU code. This should also solve the draco problem because all of draco uses the standard `Insist` and there won't be any confusion in which function is called under the hood.

Reply:
Good catch. I will add that back in. That should work because I c strings will convert to std::strings just not the other way around.